source/contributed/typescript-and-screeps-2.md

```diff
@@ -0,0 +1,598 @@
+title: TypeScript and Screeps, from Beginner to Advanced, Pt. 2
+contributed:
+    name: bryan
+    link: https://github.com/bryanbecker
+    date: 2017-05-15
+---
+
+# TypeScript and Screeps, from Beginner to Advanced, Pt. 2
+
+* [Part 1](http://docs.screeps.com/contributed/typescript-and-screeps-1.html)
+Introduction to TypeScript and setting up a TypeScript + Screeps environment
+* [Part 2](http://docs.screeps.com/contributed/typescript-and-screeps-2.html)
+**In depth look using TypeScript's features to extend the built-in Screeps objects and methods**
+* [Part 3](http://docs.screeps.com/contributed/typescript-and-screeps-3.html)
+Getters, setters, and meta programming with decorators
+
+In Part Two, we are going to take a look at using custom TypeScript interfaces and types to extend the Screeps classes to write shorter, more error-proof code.
+Towards the end of the article, we will look at some specific Screeps use-cases that take advantage of these custom types.
+
+## Advanced Typescript & Screeps Typing
+> AKA, how to supercharge your code
+
+### Extending the Default Screeps Typings with declaration merging
+
+TypeScript provides support for [declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html).  This means that declarations (most commonly, interfaces) defined throughout your code with the same name, will be merged into one large interface.  To use an example from the docs:
+
```

Comment:
I think you should use a screeps related example here, otherwise it will only duplicate the TypeScript documentation.